test/specs/library.spec.js

```diff
@@ -196,6 +196,35 @@ describe('createForm', () => {
         expect($errors.ys[0]).toBe('');
         expect($errors.ys[1]).toBe(typeError);
       });
+
+      it('handles null guard when schema is nested', async () => {
+        const initialValues = {};
+        const typeError = 'type error';
+
+        const validationSchema = yup.object().shape({
+          stringArray: yup.array().of(yup.string().typeError(typeError)),
+          objectArray: yup
+            .array()
+            .of(
+              yup
+                .object()
+                .shape({
+                  foo: yup.array().of(yup.string().typeError(typeError)),
+                }),
+            ),
+          nested: yup
+            .object()
+            .shape({foo: yup.array().of(yup.string().typeError(typeError))}),
+        });
+
+        const instance = getInstance({initialValues, validationSchema});
+
+        await instance.handleSubmit();
+        const $errors = await subscribeOnce(instance.errors);
+
+        expect($errors.nested.foo).not.toBe(typeError);
```

Comment:
we can omit the type error, not raising an error is expected. as long as we set the expectation on the output, any errors that would've been raised would fail the test

Reply:
should the expectation be removed or all typeError references?